compiler/src/dmd/astenums.d

```diff
@@ -459,3 +459,9 @@ extern (C++) struct structalign_t
     bool isPack() const    { return pack; }
     void setPack(bool pack) { this.pack = pack; }
 }
+
+/// Use to return D arrays from C++ functions
+extern (C++) struct DArray(T)
+{
+    T[] data;
+}
```

Comment:
Would it be more appropriate to put this in `root.array`? Would save adding one of the imports to cpreprocess.

Reply:
I didn't find anywhere where it belonged naturally. I figured astenums.d because it was kinda a collection of simple leaf types, and it didn't import anything else.

Reply:
root.array has a bunch of unittests, non-trivial types, and more imports.

Reply:
Seems like a good candidate for `alias this` so you don't need the extra indirection on the D side.

Reply:
> root.array has a bunch of unittests, non-trivial types, and more imports.

I suggested it because the module is already importing root.array. On the C++ side this struct of the same name is in root.dcompat.